spec/src/main/asciidoc/ecosystem.asciidoc

```diff
@@ -0,0 +1,164 @@
+= MicroProfile Config and Service Mesh
+
+MicroProfile Config provides a solution to externalise the configuration. The default config sources includes environment variables, system properties and microprofile-config.properties on the classpath. 
+
+The properties defined in Kubernetes config map can be transformed to environment variables via envFrom. After a cloud-native microservice is deployed, if an extra config source is needed, this file can be mounted to configmap, which then use envFrom facility to expose the properties to the microservie. 
+
+[source, text]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: example-config
+  namespace: default
+data:
+  example.property.1: hello
+  example.property.2: world
+  
+   
+----
+----
+# Use envFrom to load ConfigMaps into environment variables
+
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: mans-not-hot
+  labels:
+    app: mans-not-hot
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: mans-not-hot
+  template:
+    metadata:
+      labels:
+        app: mans-not-hot
+    spec:
+      containers:
+        - name: app
+          image: gcr.io/mans-not-hot/app:bed1f9d4
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 80
+          envFrom:
+          - configMapRef:
+              name: example-config
+----
+
+Therefore, config properties specified in the config map are automatically injectable to the microservices via MicroProfile Config APIs.
+
+[source, java]
+----
+@ApplicationScoped
+public class Demo {
+
+    //The property example.property.1 was defined in the configmap of Kubernetes
+    @Inject
+    @ConfigProperty(name="example.property.1") String myProp1;
+    //The property example.property.1 was defined in the configmap of Kubernetes
+    @Inject
+    @ConfigProperty(name="example.property.2") String myProp2;
+    public void echo() {
+        System.out.println(myProp1 + myProp2);
+    }
+}
+----
+The above example demonstrates that MicroProfile Config complements to service mesh via Kubernetes. 
+
+= MicroProfile Health Check and Service Mesh
+In Service Mesh architecture, each pod has a lifecycle. Service Mesh needs to know when to kill a pod and when to route requests to a pod. Therefore, it needs to know each pod's health status. A pod's health status is measured using Liveness and Readiness.
+
+=== Liveness
+Many microservices run for long periods of time and they eventually might transition to broken states, and cannot recover except being restarted. This is called liveness lifecycle.
+
+=== Readiness
+
+Sometimes, microservices are temporarily unable to serve traffic. For example, an application might need to load large data or configuration files during startup. 
+
+=== MicroProfile Health Check
+
+MicroProfile Health Check denotes whether the microservice is live or ready. It exposes an endpoint `/health`. Invoking the endpoint returns either UP (healthy) or DOWN (unhealthy).
+ 
+[source, text]
+
```

Comment:
Again, specific examples are going to get out of date quickly, especially considering the changes being discussed around separate live and ready endpoints.

Linking to the particular spec document is preferred

Reply:
updated